st2common/st2common/query/base.py

```diff
@@ -126,7 +126,9 @@ def _query_and_save_results(self, query_context, last_query_time=None):
                 LOG.debug('Removed state object %s.', query_context)
             return
 
-        if status in action_constants.LIVEACTION_COMPLETED_STATES:
+        if (status in action_constants.LIVEACTION_COMPLETED_STATES or
+                status == action_constants.LIVEACTION_STATUS_PAUSED):
```

Comment:
Hm so it looks like we will also invoke post_run when execution is paused - https://github.com/StackStorm/st2/blob/0536b072dd3a45c1cd55e827e3b8a93b23e59c85/st2common/st2common/query/base.py#L133

Is this desired?